dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/RenormalizationProcessor.java

```diff
@@ -0,0 +1,159 @@
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemProcessor;
+
+import uk.ac.ebi.eva.commons.core.models.IVariant;
+import uk.ac.ebi.eva.commons.core.models.IVariantSourceEntry;
+import uk.ac.ebi.eva.commons.core.models.VariantType;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+import uk.ac.ebi.eva.commons.core.models.pipeline.VariantSourceEntry;
+import uk.ac.ebi.eva.dbsnpimporter.sequence.FastaSequenceReader;
+
+import java.util.stream.Collectors;
+
+public class RenormalizationProcessor implements ItemProcessor<Variant, Variant> {
+
+    private static final Logger logger = LoggerFactory.getLogger(RenormalizationProcessor.class);
+
+    private FastaSequenceReader fastaSequenceReader;
+
+    public RenormalizationProcessor(FastaSequenceReader fastaSequenceReader) {
+        this.fastaSequenceReader = fastaSequenceReader;
+    }
+
+    @Override
+    public Variant process(Variant variant) throws Exception {
+        if (isAmbiguous(variant)) {
+            return renormalize(variant);
+        }
+        return variant;
+    }
+
+    /**
+     * We define the requirements to be an ambiguous variant as: being an indel (one allele is empty) and the non empty
+     * allele ends with the same nucleotide as what is in the reference assembly right before the variant.
+     *
+     * The reason for this is because we know the variants being processed have removed the leftmost nucleotide as
+     * redundant context, and we want it to have removed the rightmost nucleotide. This only yield different results
+     * if the original context nucleotide appears in the beginning and end of the other allele.
+     *
+     * Examples: let's say in position 10 there's AC, and an insertion happens between bases 10 and 11 of GGT.
+     * This is represented as 11: "" > "GGT". This won't be ambiguous because T is not the same as A (last nucleotide !=
+     * nucleotide before the variant). In contrast, if the insertion was "GGA" it is ambiguous. It would have appeared
+     * originally as 10: "A" > "AGGA", which can be interpreted as 11: "" > "GGA" (variant being processed) or as
+     * 10: "" > "AGG" (desired representation). For deletions the explanation is the same.
+     *
+     * For the needed change,
+     * @see RenormalizationProcessor#renormalize(uk.ac.ebi.eva.commons.core.models.IVariant)
+     * @see RenormalizationProcessor#renormalizeAllele(java.lang.String)
+     */
+    private boolean isAmbiguous(Variant variant) {
+        try {
+            boolean isIndel = variant.getType() == VariantType.INDEL;
+            boolean oneAlleleIsEmpty = variant.getReference().isEmpty() ^ variant.getAlternate().isEmpty();
+            return isIndel && oneAlleleIsEmpty && areContextAndLastNucleotideEqual(variant);
+        } catch (Exception e) {
+            // TODO jmmut: should we throw exception and stop the whole job if one variant is not found in the assembly? (or if there was another problem?)
```

Comment:
Variants that do not match the assembly are filtered out by another processor, so I don't think this one should care about it. The warning message looks fine to me.